src/ParallelDataCollector.cpp

```diff
@@ -956,6 +970,45 @@ namespace splash
         dataset.close();
     }
 
+    CollectionType* ParallelDataCollector::readDataSetMeta(H5Handle h5File,
+            int32_t id,
+            const char* name,
+            const Dimensions dstBuffer,
+            const Dimensions dstOffset,
+            const Dimensions srcOffset,
+            Dimensions &sizeRead,
+            uint32_t& srcDims)
+    throw (DCException)
+    {
+        log_msg(2, "readDataSetMeta");
+
+        std::string group_path, dset_name;
+        DCDataSet::getFullDataPath(name, SDC_GROUP_DATA, id, group_path, dset_name);
+
+        DCGroup group;
+        group.open(h5File, group_path);
+
+        DCDataSet dataset(dset_name.c_str());
+        dataset.open(group.getHandle());
+
+        //size_t datatype_size = 0;
+        //DCDataType dc_datatype = DCDT_UNKNOWN;
+        //datatype_size = dataset.getDataTypeSize();
+        //dc_datatype = dataset.getDCDataType();
+
+        size_t entrySize;
+        getEntriesForID(id, NULL, &entrySize);
+        std::vector<DataCollector::DCEntry> entries(entrySize);
+
+        getEntriesForID(id, &(*entries.begin()), NULL);
+
+        Dimensions src_size(dataset.getSize() - srcOffset);
+        dataset.read(dstBuffer, dstOffset, src_size, srcOffset, sizeRead, srcDims, NULL);
+        dataset.close();
+
+        return entries[0].colType;
```

Comment:
@f-schmitt-zih that's a bit hacky but if I remember correctly, should work since they all have the same type.